com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -115,9 +115,25 @@ public class NetworkSceneManager
         /// </summary>
         public event SceneEventDelegate OnSceneEvent;
 
-        internal readonly HashSet<string> RegisteredSceneNames = new HashSet<string>();
-        internal readonly Dictionary<string, uint> SceneNameToIndex = new Dictionary<string, uint>();
-        internal readonly Dictionary<uint, string> SceneIndexToString = new Dictionary<uint, string>();
+        /// Delegate declaration for the <see cref="VerifySceneBeforeLoading"/> handler that provides
+        /// an additional level of scene loading security and/or validation to assure the scene being loaded
+        /// is valid scene to be loaded in the LoadSceneMode specified.
+        /// </summary>
+        /// <param name="sceneIndex">Build Settings Scenes in Build List index of the scene</param>
+        /// <param name="sceneName">Name of the scene</param>
+        /// <param name="loadSceneMode">LoadSceneMode the scene is going to be loaded</param>
+        /// <returns>true (valid) or false (not valid)</returns>
+        public delegate bool VerifySceneBeforeLoadingDelegateHandler(int sceneIndex, string sceneName, LoadSceneMode loadSceneMode);
```

Comment:
`internal bool ValidateSceneBeforeLoading(...)` is `internal` but these types are `public`, why?

Reply:
The [RFC Explains what this is used for](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/rfc/remove-sceneregistration/text/0000-remove-sceneregistration.md).  I go into more details later in comments.